src/containerbuddy/main.go

```diff
@@ -48,10 +48,21 @@ func main() {
 		// Run our main application and capture its stdout/stderr.
 		// This will block until the main application exits and then os.Exit
 		// with the exit code of that application.
-		code, err := run(flag.Args())
+		config.Command = getCmd(flag.Args())
+		code, err := executeAndWait(config.Command)
 		if err != nil {
 			log.Println(err)
 		}
+		deregisterServices(config)
```

Comment:
> Should this be called in the top of terminate in signals.go so that the requests to the service can be stopped before the service truly stops responding to them?

If we move the call to `deregisterServices` then we'll miss cases where the application exits normally without a `docker stop` (processes don't call `SIGTERM` on themselves when they exit).

But your concern is well-founded. Maybe we need to *also* include the `deregisterServices` in the beginning of `terminate`; we'd need to either handle the error from the discovery service on the second call or set a flag that we've already terminated the process.